nifi-toolkit/nifi-toolkit-admin/src/main/groovy/org/apache/nifi/toolkit/admin/client/NiFiClientFactory.groovy

```diff
@@ -116,22 +121,27 @@ class NiFiClientFactory implements ClientFactory{
         return sslContext;
     }
 
-    private static class NiFiHostnameVerifier implements HostnameVerifier {
+    static class NiFiHostnameVerifier implements HostnameVerifier {
 
         @Override
         public boolean verify(final String hostname, final SSLSession ssls) {
             try {
-                for (final Certificate peerCertificate : ssls.getPeerCertificates()) {
-                    if (peerCertificate instanceof X509Certificate) {
-                        final X509Certificate x509Cert = (X509Certificate) peerCertificate;
-                        final List<String> subjectAltNames = getSubjectAlternativeNames(x509Cert);
-                        if (subjectAltNames.contains(hostname.toLowerCase())) {
-                            return true;
-                        }
+
+                final Certificate peerCertificate = ssls.getPeerCertificates()[0]
```

Comment:
I think this could throw an `ArrayIndexOutOfBoundsException` -- we should check the presence/size first and log/throw an exception.